### DISCUSSION TITLE
ADR004 bans webscraping, can it still be used for webpage in device firmware not being updated anymore.

### DISCUSSION THREAD

**Discussion by catsmanac** [home-assistant/architecture#952] [General] [answered]
Enphase Envoy integrations users still have legacy devices deployed with an embedded webpage to access up to 4 data items available. This data are retrieved by a simple get and regular expression, no use of beautifulsoup or PhantomJS.

Newer firmware versions for the device offer new api, but not all devices can or are being upgraded.

A new version of the Enphase Envoy integration is in the making and the question has arisen if the method of using the get with regular expression to get the 4 data for this specific firmware may be used in light of ADR004.

The driving reasons for ADR004 are addressed by using get/regex. The changing webpage is not an issue, we would actually hope for firmware updates on the device to get out of this situation but that seems unrealistic for these old devices.

**Comment by frenck** [comment, marked as answer]
This has been previously discussed when creating ADR-0004, especially around firmware that will rarely change or won't change anymore; These have also been included in ADR-0004 and are not allowed.

It means in case of Enphase Envoy, we won't be able to support those older devices in core.

../Frenck
